oidc.go

```diff
@@ -267,13 +282,43 @@ func (i *IDToken) Claims(v interface{}) error {
 	return json.Unmarshal(i.claims, v)
 }
 
+// VerifyAccessTokenHash verifies that the hash of the supplied access token
+// matches the hash in the id token. It returns an error if the hashes
+// don't match. It is the caller's responsibility to ensure that the optional
+// access token hash is present for the id token before calling this method.
+// See https://openid.net/specs/openid-connect-core-1_0.html#CodeIDToken
+func (i *IDToken) VerifyAccessTokenHash(accessToken string) error {
+	if i.AccessTokenHash == "" {
+		return ErrNoAccessTokenHash
+	}
+	var h hash.Hash
+	switch i.sigAlgorithm {
+	case RS256, ES256, PS256:
+		h = sha256.New()
+	case RS384, ES384, PS384:
+		h = sha512.New384()
+	case RS512, ES512, PS512:
+		h = sha512.New()
+	default:
+		return fmt.Errorf("VerifyAccessTokenHash: unsupported signing algorithm %q", i.sigAlgorithm)
+	}
+	h.Write([]byte(accessToken)) // hash documents that Write will never return an error
+	sum := h.Sum(nil)[:h.Size()/2]
+	actual := base64.RawURLEncoding.EncodeToString(sum)
+	if actual != i.AccessTokenHash {
+		return fmt.Errorf("invalid access token hash, want %q, got %q", i.AccessTokenHash, actual)
```

Comment:
I don't think we need to print "want" and "got" Users don't get much from a giant hash value.

Reply:
agreed but might be useful for debugging further?

Reply:
> agreed but might be useful for debugging further?

We'd need a more information than two hashes to debug further. SIG alg, access token, etc, and we shouldn't return the access token in the error message because it's sensitive.